tests/test.ts

```diff
@@ -20,6 +21,107 @@ const test_cases: {
   '々': 'cjk-letter',
 };
 
+// BMP
+const cjkRadicalsSupplement = '⻁⺪⻏';
+const kangxiRadicals = '⽒⽓⽔⽕';
+const ideographicDescriptionCharacters = '⿰⿱⿲⿳';
+const cjkSymbolsPunctuation = '「」〒〓〱〻〧〨';
+const hiragana = 'どなにぬ';
+const katakana = 'ハヒヘホ';
+const bopomofo = 'ㄬㄭㄮ';
+const alsoBopomofo = 'ㄯ';
+const hangulCompatibilityJamo = 'ㅵㅶㅷㅸ';
+const kanbun = '㆐㆑㆒㆓';
+const bopomofoExtended = 'ㆠㆡ';
+const alsoBopomofoExtendedIntroducedInUnicode13 = 'ㆾㆿ';
+const cjkStrokes = '㇀㇁㇂㇃㇄㇅㇆㇇㇈㇉㇊㇢㇣';
+const katakanaPhoneticExtensions = 'ㇼㇽㇾㇿ';
+const enclosedCjkLettersMonths = '㈱㋋㋌㋿';
+const cjkCompatibility = '㌕㍕㍖㍗';
+const alsoCjkCompatibility = '㍧㍻㍿㏄㏾';
+const cjkUnifiedIdeographsExtensionA = '䶼䶽䶾䶿';
+const hangulJamoExtendedA = 'ꥠꥹꥺꥻ';
+const hangulSyllables = '힠힡힢힣';
+const hangulJamoExtendedB = 'ퟸퟹퟺퟻ';
+const cjkCompatibilityIdeographs = '豈兀﨔隷侮恵並﨎';
+const cjkCompatibilityForms = '﹌﹍﹎﹏';
+const smallFormVariants = '﹨﹩﹪﹫';
+const halfwidthandFullwidthForms = '！Ｄ｢￥￮';
+const alsoHalfwidthandFullwidthForms = 'ﾕￇ';
+
+// SMP
+const kanaExtendedB = '𚿵𚿶𚿷𚿸';
+const kanaSupplement = '𛀀𛀁𛃾𛃿';
+const kanaExtendedA = '𛄀𛄁𛄡𛄢';
+const smallKanaExtension = '𛅕𛄲𛅤𛅥𛅦𛅧';
+const nushu = '𛋸𛋹𛋺𛋻';
+const khitanSmallScript = '𘳒𘳓𘳔𘳕';
+const cjkUnifiedIdeographsExtensionB = '𪛜𪛝𪛞𪛟';
+const cjkUnifiedIdeographsExtensionC = '𫚄𫚅𫜸𫜹';
+const cjkUnifiedIdeographsExtensionD = '𫝀𫝁𫝂𫝃';
+const cjkUnifiedIdeographsExtensionE = '𬷰𬷱𬺠𬺡';
+const cjkUnifiedIdeographsExtensionF = '𬻿𮨥𮨦𮯠';
+const cjkUnifiedIdeographsExtensionG = '𰀀𰀁𱍀𱍊';
+const cjkUnifiedIdeographsExtensionH = '𱍐𱍑𲎬𲎭';
+const cjkCompatibilityIdeographsSupplement = '乁你鼻𪘀';
+
+function casify(chars: string, t: TestType): TestCases {
+  return Object.fromEntries(chars.split('').map(c => [c, t]));
+}
+
+const more_test_cases: TestCases = {
+  ...casify(cjkRadicalsSupplement, 'cjk-letter'),
+  ...casify(cjkSymbolsPunctuation.slice(0, 5), 'cjk-punctuation'),
+  ...casify(cjkSymbolsPunctuation.slice(6, 8), 'cjk-letter'),
```

Comment:
Is it expected that the character in index 5 is skipped?

Reply:
A vivid mistake demonstrating that `.slice` should be avoided here.